draft-ietf-rats-msg-wrap.md

```diff
@@ -319,27 +349,73 @@ with the following wire representation:
 
-Layered attesters and composite devices ({{Sections 3.2 and 3.3 of -rats-arch}}) generate evidence that consists of multiple parts.
+There are cases where CMW and CMW collection payloads need to be transported in PKIX messages, for example in Certificate Signing Requests (CSR) {{-csr-a}}, or in X.509 Certificates and Certificate revocation lists (CRL) {{DICE-arch}}.
 
-For example, in data center servers, it is not uncommon for separate attesting environments (AE) to serve a subsection of the entire machine.
-One AE might measure and attest to what was booted on the main CPU, while another AE might measure and attest to what was booted on a SmartNIC plugged into a PCIe slot, and a third AE might measure and attest to what was booted on the machine's GPU.
+For CMW, Section 6.1.8 of {{DICE-arch}} defines the ConceptualMessageWrapper format and the associated extension object identifier.
 
-To address the composite Attester use case, this document defines a CMW "collection" as a container that holds several CMW evidence conceptual messages, each with a label that is unique within the scope of the collection.
+This section specifies the CMWCollection extension to carry CMW collection objects.
 
-The CMW collection ({{fig-cddl-collection}}) is defined as a CBOR map or JSON object with CMW values.
-The position of a `cmw` entry in the `cmw-collection` is not significant.
-Instead, the labels identify a conceptual message that corresponds to a component of a system.
-Labels can be strings or integers that serve as a mnemonic for different conceptual messages in the collection.
+The CMWCollection extension MAY be included in X.509 Certificates, CRLs {{-pkix}}, and CSRs.
 
-~~~ cddl
-{::include cddl/cmw-collection.cddl}
+The CMWCollection extension MUST be identified by the following object identifier:
+
+~~~asn.1
+id-pe-cmw-collection  OBJECT IDENTIFIER ::=
+        { iso(1) identified-organization(3) dod(6) internet(1)
+          security(5) mechanisms(5) pkix(7) id-pe(1) TBD }
 ~~~
-{: #fig-cddl-collection artwork-align="left"
-   title="CDDL definition of the CMW collection format"}
 
-Although initially designed for the composite attester use case, the CMW collection can be repurposed for other use cases requiring CMW aggregation.
+This extension MUST NOT be marked critical.
+
+The CMWCollection extension MUST have the following syntax:
+
+~~~asn.1
+CMWCollection ::= OCTET STRING
+~~~
+
+The CMWCollection MUST contain the JSON or CBOR serialization of a CMW collection object.
+
+## ASN.1 Module {#asn1-x509}
+
+This section provides an ASN.1 module {{X.680}} for the CMWCollection extension, following the conventions established in {{RFC5912}} and {{RFC6268}}.
+
+
+~~~asn.1
+CMWCollectionExtn
+  { iso(1) identified-organization(3) dod(6) internet(1)
+    security(5) mechanisms(5) pkix(7) id-mod(0)
+    id-mod-cmw-collection-extn(TBD) }
+
+DEFINITIONS IMPLICIT TAGS ::=
+BEGIN
+
+IMPORTS
+  EXTENSION
+  FROM PKIX-CommonTypes-2009  -- RFC 5912
```

Comment:
This appears to be boilerplate. Should this be explained better? Seems to leave the reader guessing at its significance.

Reply:
That's how the PKIX registration machinery is set up.